### PR TITLE
[#4] [KMM] Set Up CICD for KMM

### PR DESCRIPTION
**Pull request by blyscuit**
## What Happened

- Use Kover for unit test and test coverage.
- Use Detekt for lint.
- Display Kover and Detekt result via Danger.
- Update current CI workflow for `android_pull_request`.

## Insight

- Use Danger as Gradle Task to support configuration file in Kotlin.

## Proof of Work

Screen Shot 2022-09-16 at 15 43 11

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import org.gradle.kotlin.dsl.`kotlin-dsl`
+
+plugins {
+    `kotlin-dsl`
+}
+
+repositories {
+    mavenCentral()
+}
```
Comment (suho) [inline buildSrc/build.gradle.kts]: @blyscuit You can setup the android studio to always have the break line at the end

File: Dangerfile [hunk 2]
```diff
@@ -14,26 +14,26 @@ warn("Please add labels to this PR") if github.pr_labels.empty?
 commit_lint.check warn: :all, disable: [:subject_length]
 
 # Detekt output check
-detekt_dir = "**/build/reports/detekt/detekt-result.xml" 
+detekt_dir = "**/build/reports/detekt/detekt.xml"
```
Comment (minhnimble) [inline Dangerfile]: we have the convention to add the postfix `-result` in the Android Template for the Danger, not sure why you remove it? Shouldn't we generate the detekt file report with that prefix? 🤔

File: androidApp/build.gradle.kts [hunk 3]
```diff
@@ -4,11 +4,11 @@ plugins {
...
 
 android {
-    compileSdk = 32
+    compileSdk = Android.CompileSdk
     defaultConfig {
         applicationId = "co.nimblehq.blisskmmic.android"
-        minSdk = 26
-        targetSdk = 32
+        minSdk = Android.MinSdk
+        targetSdk = Android.CompileSdk
```
Comment (minhnimble) [inline androidApp/build.gradle.kts]: Small naming refactor suggestion:
```suggestion
        targetSdk = Android.TargetSdk
```

File: buildSrc/build.gradle.kts (hunk 1 above)
Comment (minhnimble) [inline buildSrc/build.gradle.kts]: Missing `end of line` (eol) for this file

File: buildSrc/src/main/kotlin/myPackage/Versions.kt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+object Android {
+    const val CompileSdk = 32
+    const val MinSdk = 26
+}
```
Comment (minhnimble) [inline buildSrc/src/main/kotlin/myPackage/Versions.kt]: Ditto here on missing eol

File: shared/build.gradle.kts [hunk 5]
```diff
@@ -58,10 +61,33 @@ kotlin {
...
 
 android {
-    compileSdk = 32
+    compileSdk = Android.CompileSdk
     sourceSets["main"].manifest.srcFile("src/androidMain/AndroidManifest.xml")
     defaultConfig {
-        minSdk = 26
-        targetSdk = 32
+        minSdk = Android.MinSdk
+        targetSdk = Android.CompileSdk
```
Comment (minhnimble) [inline shared/build.gradle.kts]: Ditto here on naming refactor:
```suggestion
        targetSdk = Android.TargetSdk
```

**Comment by blyscuit** [comment]
@minhnimble @suho Fixed issues with line breaks and naming in https://github.com/blyscuit/kmm-swiftUI-IC/pull/48/commits/9fb1826287eef3d9a802ec83bc0175eb7f2b4a6f

File: Dangerfile (hunk 2 above)
Comment (blyscuit) [reply Dangerfile]: Good suggestion.

**Review by minhnimble — APPROVED** [review]

**Review by suho — APPROVED** [review]
LGTM!

**Pull request merged**